python/PEDvPUDplot.py

```diff
@@ -2,104 +2,166 @@
+    
+    Order of list elements corresponds to order of distances (left to right) in each matrix row (top to bottom) in the SDT file. Pairwise identities of 0 between the same pairwise sequences are excluded. Pairwise distances are calculated as the complement of identities.
+    
+    ::param filepath::     Path to SDT file (input).
+    ::param sdtsims::      Input file.
+    ::param sdtdist::      List of pairwise distances (float).
+    ::param line::         Line from input file.
+    ::param simsline::     List of elements from param line.
+    ::param similarities:: Slice of param simsline to exclude sequence ID & distance of 0 (pairwie comparison with self).
+    ::param item::         Individual distance measure.
+    """
     with open(filepath) as sdtsims:
         sdtdist = []
-        for lines in sdtsims:
-            simsline = lines.split(',')
+        for line in sdtsims:
+            simsline = line.split(',')
             similarities = simsline[1:(len(simsline)-1)]
             for item in similarities:
                 item = item.strip('\n')
                 item = float(item)
-                sdtdist.append(comp(item))
+                sdtdist.append(1-item)
     return sdtdist    
 
-# function to get pairwise uncorrected distances from SDT output file
-# key (tuples of sequence IDs): value (comp(SDT identities))
 def getPUD(filepath):
+    """
+    Calls functions getSDTid and getSDTdist and returns dictionary of pairwise sequence (tuple) keys and pairwise uncorrected distance values.
+    
+    ::param filepath::     Path to SDT file (input).
+    ::param SDTid::       List of sequence ID strings.
+    ::param SDTdist::     List of dictionary values (float distances).
+    ::param seqpairs::    List of dictionary keys (tuple of sequence IDs).
+    ::param seq1Count::   Var to keep count of 1st sequence ID in pair.
+    ::param seq2Count::   Var to keep count of 2nd sequence ID in pair.
+    ::param eachseqpair:: Tuple of sequence IDs
+    ::param PUDS::        Dictionary of pairwise sequence keys and uncorrected distance values, excluding pairwise comparisons of sequence with itself.
+    """
     SDTid = getSDTid(filepath)
-    SDTdist = getSDTdist(filepath) # distance values
-    seqpairs = [] # tuple keys
-    i = 1
-    while i<len(SDTid):
-        j = 0
-        while j < i:
-            eachseqpair = (SDTid[i], SDTid[j])
+    SDTdist = getSDTdist(filepath) 
+    seqpairs = [] 
+    seq1Count = 1
+    while seq1Count<len(SDTid):
+        seq2Count = 0
+        while seq2Count < seq1Count:
+            eachseqpair = (SDTid[seq1Count], SDTid[seq2Count])
             seqpairs.append(eachseqpair)
-            j += 1
-        i += 1
+            seq2Count += 1
+        seq1Count += 1
     PUDS = dict(itertools.izip(seqpairs, SDTdist))
     return PUDS
 
-# function to get a list of lists comprising sequence ID and TreePuzzle distances
 def getTPZiddist(filepath2):
```

Comment:
Also, 
* Where is the python documentation for this?
* Does it make sense to have distances between Ids?  Should this be perhaps getTpzDistance?